### ISSUE TITLE
Which of the 2019 SOTA models can be used with simple_streaming_asr_example?

### ISSUE THREAD

**Issue by abhinavkulkarni** [facebookresearch/wav2letter#710]
Hi,

Which of the [2019 SOTA models](https://github.com/facebookresearch/wav2letter/tree/master/recipes/models/sota/2019) can be used with `simple_streaming_asr_example` and `interactive_streaming_asr_example` binaries from inference framework?

The SOTA models seem to vary along the following dimensions:
1. Architecture (ResNet, TDS, Transformer)
2. Loss function (Seq2Seq, CTC)
3. Tokens (word or word-piece based)
4. LM (NGram or DNN based)

Can you please comment along these 4 dimensions, which of the SOTA models can be used with the inference framework?

Thanks for the great work!

**Comment by tlikhomanenko** [comment]
@vineelpratap @avidov please correct me if I am wrong:

1. ResNet and TDS models
2. CTC
3. AM can use any word-pieces, LM should be word-based (possibly we can also run with lexicon token-based decoding)
4. Ngram

**Comment by vineelpratap** [comment]
Regarding (1), we support only TDS automatically for now. See, https://github.com/facebookresearch/wav2letter/tree/master/recipes/models/streaming_convnets for example architecture.

**Comment by abhinavkulkarni** [comment]
Hey @tlikhomanenko, @vineelpratap:

I am having trouble running the `simple_streaming_example` binary with the TDS CTC acoustic model. 

Firstly, I am able to successfully run the model using the example on the Wiki:

```bash
:$ cat /root/host/audio/JRE_Elon_Musk_#1470_10s.wav | \
 /root/wav2letter/build/inference/inference/examples/simple_streaming_asr_example \
 -acoustic_module_file wav2letter/acoustic_model.bin \
 -feature_module_file wav2letter/feature_extractor.bin \
 -language_model_file wav2letter/language_model.bin \
 -lexicon_file wav2letter/lexicon.txt \
 -tokens_file wav2letter/tokens.txt \
 -decoder_options_file wav2letter/decoder_options.json
```
I tried to run the above using following command (I changed the am, lm, lexicon and token params to use different ones):

```bash
:$ cat /root/host/audio/JRE_Elon_Musk_#1470_10s.wav | \
 /root/wav2letter/build/inference/inference/examples/simple_streaming_asr_example \
 -acoustic_module_file sota/am/am_tds_ctc_librivox_dev_other.bin \
 -feature_module_file wav2letter/feature_extractor.bin \
 -language_model_file sota/lm/lm_librispeech_kenlm_word_4g_200kvocab.bin \
 -lexicon_file sota/lexicon/decoder-unigram-10000-nbest10.lexicon \
 -tokens_file sota/tokens/librispeech-train-all-unigram-10000.tokens \
 -decoder_options_file wav2letter/decoder_options.json
```

I get the following error: 
```bash
Started features model file loading ... 
Completed features model file loading elapsed time=10527 microseconds

Started acoustic model file loading ... 
terminate called after throwing an instance of 'cereal::Exception'
  what():  Error while trying to deserialize a polymorphic pointer. Could not find type id 3
Aborted (core dumped)
```

**Comment by tlikhomanenko** [comment]
cc @vineelpratap

**Comment by chrismbirmingham** [comment]
Can you provide instructions for testing other pretrained models with simple_streaming_asr_example? What configurations need to be changed (if any) and where to find the download link for the models as described in the wiki:

`for f in acoustic_model.bin tds_streaming.arch decoder_options.json feature_extractor.bin language_model.bin lexicon.txt tokens.txt ; do wget http://dl.fbaipublicfiles.com/wav2letter/inference/examples/model/${f} ; done` 

I've gotten it working with this default but tbh I am not actually sure what paper this model comes from.

Any help/advice is greatly appreciated!

**Comment by tlikhomanenko** [comment]
cc @vineelpratap

**Comment by MrityunjoyS** [comment]
Hi, @abhinavkulkarni, I'm also getting the same error as you did. Can you please suggest if you're able to find a workaround.

**Comment by abhinavkulkarni** [comment]
@chrismbirmingham: The model on the inference wiki comes from `/wav2letter/recipes/models/streaming_convnets/librispeech` recipe. The model on the wiki (which can be run in a streaming fashion) was obtained by converting the one obtained from the above recipe by the converter tool `/wav2letter/tools/StreamingTDSModelConverter.cpp`.

**Comment by abhinavkulkarni** [comment]
@MrityunjoyS: I haven't revisited the problem since I made this issue, but it could be that the model needs to be converted to a streaming one using the converter utility: `/wav2letter/tools/StreamingTDSModelConverter.cpp`.
